runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java

```diff
@@ -146,7 +144,7 @@ public String getJobId() {
    * Get the project this job exists in.
    */
   public String getProjectId() {
```

Comment:
It is used in this class's other functions.
But, it is public, and could be useful for users. So, I am not confident for deleting it.

I agree with the mutation issue. However, dataflowOptions is also used for MonitoringUtil.getGcloudCancelCommand(). The current change will make the code consistent at least.